milestone-3/front-end/src/components/recipePage.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import './recipePage.css';
+
+export default function RecipePage(props) {
+   const [newIngredient, setNewIngredient] = useState('');
+
+   return (
+      <main id="main">
+         <h1 id="recipeName">{ props.recipeName }</h1>
+         <h3 id="recipeSubtitle">{ props.recipeSubtitle }</h3>
+         <p id="recipeDescription">{ props.recipeDescription }</p>
+         
+         <img id="recipeImage" className="recipeImage" src={ props.recipeImage } alt={ props.imageAlt } />
+
+         <h2>Ingredients</h2>
+         <ul id="ingredientList">
+            {
+               props.ingredientList.map(function(name, index) {
+                  return <li key={ index }>{ name }</li>;
+               })
+            }
+         </ul>
+         <form>
+            <input
+               type="text"
+               id="inputIngredient"
+               placeholder="e.g. 1/2 cup of flour"
+               value={newIngredient}
+               onChange={(e) => {
+                  setNewIngredient(e.target.value);
+               }}
+            />
+            <button type="button" onClick={ () => console.log(newIngredient) }>
+               Add ingredient
+            </button>
```

Comment:
So close! You are updating state correctly, so all you need to do now is copy your addIngredients function from milestone 2 into this component and remove the `let newIngredient = document.getElementById("inputIngredient").value;` line since newIngredient is now stored in the state variable

Reply:
Got it, thank you for the feedback! I'll update this to make it functional.